data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/DefaultPluginFactory.java

```diff
@@ -109,12 +113,23 @@ private <T> PluginArgumentsContext getConstructionContext(final PluginSetting pl
     }
 
     private <T> Class<? extends T> getPluginClass(final Class<T> baseClass, final String pluginName) {
-        return pluginProviders.stream()
+        final Class<? extends T> pluginClass = pluginProviders.stream()
                 .map(pluginProvider -> pluginProvider.findPluginClass(baseClass, pluginName))
                 .filter(Optional::isPresent)
                 .map(Optional::get)
                 .findFirst()
                 .orElseThrow(() -> new NoPluginFoundException(
                         "Unable to find a plugin named '" + pluginName + "'. Please ensure that plugin is annotated with appropriate values."));
+
+        logDeprecatedPluginsNames(pluginClass, pluginName);
+        return pluginClass;
+    }
+
+    private <T> void logDeprecatedPluginsNames(final Class<? extends T> pluginClass, final String pluginName) {
```

Comment:
Please add test case for this. I think it is OK if it does not assert the logging. But, it should verify that we still get the plugin and nothing fails.